cli/ballerina-cli/src/main/java/io/ballerina/cli/task/CompileTask.java

```diff
@@ -208,16 +213,41 @@ public void execute(Project project) {
             if (project.buildOptions().dumpBuildTime()) {
                 BuildTime.getInstance().codeGenDuration = System.currentTimeMillis() - start;
             }
+            // HashSet to keep track of the diagnostics to avoid duplicate diagnostics
+            HashSet<String> diagnosticSet = new HashSet<>();
```

Comment:
```suggestion
            Map<String> diagnosticSet = new HashSet<>();
```
Use the abstract type for the LHS when declaring variables.